src/app/introduction/activityintro/activityintro.component.spec.ts

```diff
@@ -1,25 +1,43 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {SharedDataService} from '../../services/shared.data.service';
+import {APIService} from '../../services/api.service';
+import {DashboardService} from '../../services/dashboard.service';
+import {ActivityintroComponent} from './activityintro.component';
+import {HttpClientModule} from '@angular/common/http';
+import {LanguageService} from '../../services/language.service';
+import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {ToastrService, ToastrModule} from 'ngx-toastr';
 
-import { ActivityintroComponent } from './activityintro.component';
 
 describe('ActivityintroComponent', () => {
-  let component: ActivityintroComponent;
-  let fixture: ComponentFixture<ActivityintroComponent>;
+    let component: ActivityintroComponent;
+    let fixture: ComponentFixture<ActivityintroComponent>;
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      declarations: [ ActivityintroComponent ]
-    })
-    .compileComponents();
-  }));
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [HttpClientModule, HttpClientTestingModule, ToastrModule.forRoot()],
+            declarations: [ ActivityintroComponent ],
+            providers: [
+                // { provide: Http, useValue: mockHttpProvider },
+                SharedDataService,
+                DashboardService,
+                APIService,
+                LanguageService,
+                ToastrService
+            ]
+        })
+        .compileComponents();
+    }));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(ActivityintroComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
+    beforeEach(() => {
+        fixture = TestBed.createComponent(ActivityintroComponent);
+        component = fixture.componentInstance;
```

Comment:
Check the Indentation..  use spaces. I think 2 spaces but check the style guide for a standard.